src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -877,18 +872,11 @@ private Task WriteToStreamAsync(byte[] buffer, int offset, int count, Cancellati
 
                 // If we were unable to get more, it's an error.
                 // Otherwise, loop around to look again.
-                int newRemaining = _readLength - _readOffset;
-                if (remaining == newRemaining)
-                {
-                    break;
-                }
-                remaining = newRemaining;
+                remaining = _readLength - _readOffset;
```

Comment:
The way this is written, `remaining` is going to be lifted to the state machine.  Can we just get rid of it entirely and replace `if (remaining >= 2)` with `if (_readLength - _readOffset >= 2)`, or at the very least move the `int remaining = _readLength - _readOffset;` to inside the while block and remove this line from here?